app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
       self._importRequiredSearchParam('scope', 'scope');
       self.importSearchParam('state');
-      self.importSearchParam('redirect_uri', 'redirectUri');
+      self._importValidateOptionalSearchParam('redirect_uri', 'redirectUri', Validate.isUriValid);
       self.importSearchParam('access_type', 'accessType');
-      self.importSearchParam('redirectTo');
-      self.importSearchParam('verification_redirect', 'verificationRedirect');
+      self._importValidateOptionalSearchParam('redirectTo', 'redirectTo', Validate.isUriValid);
+      self._importValidateOptionalSearchParam('verification_redirect', 'verificationRedirect', Validate.isUriValid);
```

Comment:
Not for this PR, but it's really beginning to seem like this could be done in a declarative fashion:

```js
importSearchParams: {
   client_id: 'clientId',
   scope: 'scope',
   state: {
     dest: 'state',
     optional: true
   },
   redirect_uri: {
     dest: 'redirectUri',
     validation: Validate.isUriValid
   },
   ....
}
```